src/NumericalAlgorithms/Integration/GslQuadAdaptive.hpp

```diff
@@ -150,43 +111,51 @@ class GslQuadAdaptive;
  * functions, whereas a lower-order rule saves time for functions with local
  * difficulties, such as discontinuities.
  */
-template <typename... Args>
-class GslQuadAdaptive<IntegralType::StandardGaussKronrod, Args...>
-    : public detail::GslQuadAdaptiveImpl<Args...> {
+template <>
+class GslQuadAdaptive<GslIntegralType::StandardGaussKronrod>
+    : public detail::GslQuadAdaptiveImpl {
  public:
-  using detail::GslQuadAdaptiveImpl<Args...>::GslQuadAdaptiveImpl;
-  Result operator()(double lower_boundary, double upper_boundary,
-                    double tolerance_abs, int key,
-                    double tolerance_rel = 0.) const noexcept {
-    Result result;
-    int status = gsl_integration_qag(
-        &(this->integrand_), lower_boundary, upper_boundary, tolerance_abs,
-        tolerance_rel, this->max_intervals_, key, this->workspace_.get(),
-        &result.value, &result.error);
+  using detail::GslQuadAdaptiveImpl::GslQuadAdaptiveImpl;
+  template <typename IntegrandType>
+  double operator()(IntegrandType&& integrand, double lower_boundary,
+                    double upper_boundary, double tolerance_abs, int key,
+                    double tolerance_rel = 0.) const {
+    double result;
+    double error;
+    detail::disable_gsl_error_handling();
+    const int status_code = gsl_integration_qag(
+        gsl_integrand(std::forward<IntegrandType>(integrand)), lower_boundary,
+        upper_boundary, tolerance_abs, tolerance_rel, this->max_intervals_, key,
+        this->workspace_.get(), &result, &error);
+    detail::check_status_code(status_code);
     return result;
   }
 };
 
-/*!
- * The algorithm for "IntegrableSingularitiesPresent" concentrates new,
- * increasingly smaller subintervals around an unknown singularity and makes
- * successive approximations to the integral which should converge towards a
- * limit. The integration region is defined by `lower_boundary` and
- * `upper_boundary`.
- */
-template <typename... Args>
-class GslQuadAdaptive<IntegralType::IntegrableSingularitiesPresent, Args...>
-    : public detail::GslQuadAdaptiveImpl<Args...> {
+// /*!
+//  * The algorithm for "IntegrableSingularitiesPresent" concentrates new,
+//  * increasingly smaller subintervals around an unknown singularity and makes
+//  * successive approximations to the integral which should converge towards a
+//  * limit. The integration region is defined by `lower_boundary` and
+//  * `upper_boundary`.
+//  */
```

Comment:
Remove the extra `//` in front of these lines

Reply:
Sorry about that.